pkg/webhooks/validation_test.go

```diff
@@ -0,0 +1,543 @@
+				{
+					"apiVersion": "v1",
+					"kind": "Pod",
+					"metadata": {
+					   "name": "test-pod",
+					   "namespace": "test"
+					},
+					"spec": {
+					   "containers": [
+						  {
+							 "name": "nginx",
+							 "image": "nginx:latest"
+						  }
+					   ]
+					}
+				 }
+			`),
+			blocked: false,
+		},
+		{
+			rawPolicy: []byte(`
+				{
+					"apiVersion": "kyverno.io/v1",
+					"kind": "ClusterPolicy",
+					"metadata": {
+					   "name": "check-label-app"
+					},
+					"spec": {
+					   "validationFailureAction": "enforce",
+					   "validationFailureActionOverrides": 
+							[
+								{
+									"action": "enforce",
+									"namespaces": [
+										"default"
+									]
+								},
+								{
+									"action": "audit",
+									"namespaces": [
+										"test"
+									]
+								}
+							],
+					   "rules": [
+						  {
+							 "name": "check-label-app",
+							 "match": {
+								"resources": {
+								   "kinds": [
+									  "Pod"
+								   ]
+								}
+							 },
+							 "validate": {
+								"message": "The label 'app' is required.",
+								"pattern": {
+									"metadata": {
+										"labels": {
+											"app": "?*"
+										}
+									}
+								}
+							}
+						  }
+					   ]
+					}
+				 }
+		 	`),
+			rawResource: []byte(`
+				{
+					"apiVersion": "v1",
+					"kind": "Pod",
+					"metadata": {
+					   "name": "test-pod",
+					   "namespace": "default"
+					},
+					"spec": {
+					   "containers": [
+						  {
+							 "name": "nginx",
+							 "image": "nginx:latest"
+						  }
+					   ]
+					}
+				 }
+			`),
+			blocked: true,
+		},
+		{
+			rawPolicy: []byte(`
+				{
+					"apiVersion": "kyverno.io/v1",
+					"kind": "ClusterPolicy",
+					"metadata": {
+					   "name": "check-label-app"
+					},
+					"spec": {
+					   "validationFailureAction": "enforce",
+					   "validationFailureActionOverrides": 
+							[
+								{
+									"action": "enforce",
+									"namespaces": [
+										"default"
+									]
+								},
+								{
+									"action": "audit",
+									"namespaces": [
+										"test"
+									]
+								}
+							],
+					   "rules": [
+						  {
+							 "name": "check-label-app",
+							 "match": {
+								"resources": {
+								   "kinds": [
+									  "Pod"
+								   ]
+								}
+							 },
+							 "validate": {
+								"message": "The label 'app' is required.",
+								"pattern": {
+									"metadata": {
+										"labels": {
+											"app": "?*"
+										}
+									}
+								}
+							}
+						  }
+					   ]
+					}
+			 	}
+			`),
+			rawResource: []byte(`
+				{
+					"apiVersion": "v1",
+					"kind": "Pod",
+					"metadata": {
+					   "name": "test-pod",
+					   "labels": {
+						   "app": "my-app"
+					   }
+					},
+					"spec": {
+					   "containers": [
+						  {
+							 "name": "nginx",
+							 "image": "nginx:latest"
+						  }
+					   ]
+					}
+				 }
+			`),
+			blocked: false,
```

Comment:
Assume the namespace is `default`, so it should be blocked.

Reply:
No, since we have the 'app' label.